evalml/tests/component_tests/test_components.py

```diff
@@ -1804,3 +1807,26 @@ def test_component_parameters_supported_by_list_API(component_class):
         assert not component_class._supported_by_list_API
     else:
         assert component_class._supported_by_list_API
+
+
+def test_handle_nullable_types(
+    test_classes,
+    nullable_type_test_data,
+    nullable_type_target,
+):
+    MockComponent, _, _ = test_classes
+    y = nullable_type_target(ltype="IntegerNullable", has_nans=False)
+    X = nullable_type_test_data(has_nans=False)
```

Comment:
100% agree! I will look into the best way to test. Right now, I'm just thinking that it might be better to make a separate fixture that can take inputs instead of further bastardizing the `test_classes` like I am right now, which are used for much much simpler tests elsewhere in this file.